test/auth.e2e-spec.ts

```diff
@@ -0,0 +1,159 @@
+import { Sequelize } from 'sequelize-typescript';
+import { Entities } from 'src/app.module';
+import { AuthModule } from 'src/auth/auth.module';
+import { SingInAuthUserDto } from 'src/auth/dto/create/create-auth.dto';
+import { User } from 'src/auth/entities/user.entity';
+import { RegistrationMethod } from 'src/auth/enums/registMethod-enum';
+import { SendCodeService } from 'src/auth/services/sendCode.service';
+import { AccessTokenGuard } from 'src/common/guards/jwt/accessToken.guard';
+import { AccessTokenStrategy } from 'src/common/strategies/accessToken.strategy';
+import { RefreshTokenStrategy } from 'src/common/strategies/refreshToken.strategy';
+import * as request from 'supertest';
+
+describe('Auth', () => {
+  let app: INestApplication;
+  let sequelize: Sequelize;
+  let createdUser: User;
+  let sendCodeService: SendCodeService;
+  let accessToken: string;
+  let refreshToken: string;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        ConfigModule.forRoot({
+          envFilePath: '.env',
+          isGlobal: true,
+        }),
+        MailerModule.forRoot({
+          transport: {
+            host: process.env.MAIL_HOST,
+            port: process.env.MAIL_PORT,
+            secure: false,
+            auth: {
+              user: process.env.MAIL_USER,
+              pass: process.env.MAIL_PASS,
+            },
+          },
+        }),
+        BullModule.forRoot({
+          connection: {
+            host: process.env.REDIS_HOST,
+            port: parseInt(process.env.REDIS_PORT, 10) || 6379,
+          },
+        }),
+        SequelizeModule.forRoot({
+          dialect: 'postgres',
+          host: 'localhost',
+          port: parseInt(process.env.POSTGRES_PORT, 10) || 5432,
+          username: process.env.POSTGRES_USER,
+          password: process.env.POSTGRES_PASSWORD,
+          database: process.env.POSTGRES_DB,
+          schema: 'store',
+          models: [...Entities],
+          pool: {
+            max: 10,
+            min: 3,
+          },
+          autoLoadModels: true,
+          synchronize: true,
+        }),
+        AuthModule,
+      ],
+      providers: [
+        RefreshTokenStrategy,
+        AccessTokenStrategy,
+        {
+          provide: APP_GUARD,
+          useClass: AccessTokenGuard,
+        },
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    sequelize = moduleFixture.get<Sequelize>(Sequelize);
+    sendCodeService = moduleFixture.get<SendCodeService>(SendCodeService);
+
+    await app.init();
+  });
+
+  it('/POST auth', async () => {
+    const user: SingInAuthUserDto = {
+      email: 'da22asasds2a@gmail.com',
+      password: 'Example123456!',
+      passwordRepeat: 'Example123456!',
+      registrationMethod: RegistrationMethod.EMAIL,
+    };
+
+    jest.spyOn(sendCodeService, 'send').mockImplementation(async () => {});
```

Comment:
If I'm not mistaken, after the login application will send some code to the user. And proper test will be when you check that a related method of that service has been called. You don't need to check that the email is delivered, just that your application sends it. 

Something similar like this 

```ts
const sendEmail = jest.spyOn(sendCodeService, 'send').mockImplementation(async () => {});
expect(sendEmail).toHaveBeenCalled()
```